app/knights_config/character_configuration.py

```diff
@@ -0,0 +1,20 @@
+class Character:
+
+    def __init__(self, name_knight: str, knight: dict) -> None:
+        self.name_knight = name_knight
+        self.knight = knight
+
+    def new_character(self) -> dict:
+        hero = self.knight[self.name_knight]
+        hero["protection"] = 0
+        for armour in hero["armour"]:
+            hero["protection"] += armour["protection"]
+        hero["power"] += hero["weapon"]["power"]
+        if hero["potion"] is not None:
+            if "power" in hero["potion"]["effect"]:
+                hero["power"] += hero["potion"]["effect"]["power"]
+            if "protection" in hero["potion"]["effect"]:
+                hero["protection"] += hero["potion"]["effect"]["protection"]
+            if "hp" in hero["potion"]["effect"]:
+                hero["hp"] += hero["potion"]["effect"]["hp"]
+        return hero

```

Comment:
It will be better if you parse the knight dict and create instance attributes like protection, hp, etc. And then you can just use these attributes instead of dict keys.